timmyichen/dev-directory
# Add Onboarding page and redirect from login via auth-redirect

Closes #76 

# Description

- add redirect from `/auth-redirect` to `/onboarding` instead of the previous `/`
- Add Onboarding page with skip button
- skip button takes you back to `/` (temporary till `/directory` exists )

## Testing

To Manually test
1. ` npm run dev:server` w/ docker already running
2. Login on `/` route
3. Authorize Discord
4. Will be redirected from oAuth page to `/auth-redirect`
5. On `/auth-redirect` route after verifying the user is authorized via a `router.query.code` existing, page will redirect to `/onboarding` instead of the previous `/`
6. verify you see the onboarding page
7. click skip
8. will be redirected to home page with this url `/` (temporary till `/directory` exists )
9. testing done

## Type of change

- New feature (non-breaking change which adds functionality)

## Screenshots
`/onboarding/`
![image](https://user-images.githubusercontent.com/68509315/230419962-74a8998a-6e74-47a5-9587-3bbc5a4bd98e.png)


# Checklist:

- [ ] Changes have new/updated automated tests, if applicable
- [ ] Changes have new/updated docs, if applicable
- [x] I have added comments on any new, hard-to-understand code
- [x] Changes generate no new errors or warnings
